---
title: "Deseq2LungCancer"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Hisat2 used to align the fasta file to reference sequence to produce .bam mapped reads

Kim, D., Paggi, J.M., Park, C. et al. Graph-based genome alignment and genotyping with HISAT2 and HISAT-genotype. Nat Biotechnol 37, 907–915 (2019). https://doi.org/10.1038/s41587-019-0201-4
Human index used to map sequence:
 https://genome-idx.s3.amazonaws.com/hisat/grch38_genome.tar.gz
 genome then built using hisat2-build genome.fa index_name
(will include transcripts variants and SNPs to further increase detail it wont produce major changes but might increase accuracy for specific pathways. other tools like salmon can also be used as they map directly to transcript sequence) Can look into this further if there is a specific interest.

Output: 93.8% aligned concordantly exactly 1 time 
98.92% overall alignment rate
both are indicators of good quality

Rsubread used to produce count matrices
```{r}
#library(Rsubread)

#Count_1 <- featureCounts(files="Data/1.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_2 <- featureCounts(files="Data/2.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_3 <- featureCounts(files="Data/3.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_4 <- featureCounts(files="Data/4.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_5 <- featureCounts(files="Data/5.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_6 <- featureCounts(files="Data/6.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_7 <- featureCounts(files="Data/7.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_8 <- featureCounts(files="Data/8.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_9 <- featureCounts(files="Data/9.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_10 <- featureCounts(files="Data/10.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_11 <- featureCounts(files="Data/11.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
#Count_12 <- featureCounts(files="Data/12.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")

```


Start from loading the count matrices produced in other script
```{r message = FALSE, warning = FALSE}
Count_1 <- readRDS(file = "Count_1")
Count_2 <- readRDS(file = "Count_2")
Count_3 <- readRDS(file = "Count_3")
Count_4 <- readRDS(file = "Count_4")
Count_5 <- readRDS(file = "Count_5")
Count_6 <- readRDS(file = "Count_6")
Count_7 <- readRDS(file = "Count_7")
Count_8 <- readRDS(file = "Count_8")
Count_9 <- readRDS(file = "Count_9")
Count_10 <- readRDS(file = "Count_10")
Count_11 <- readRDS(file = "Count_11")
Count_12 <- readRDS(file = "Count_12")



```

```{r message = FALSE, warning = FALSE}
countdata <- data.frame(Count_1$counts, Count_2$counts, Count_3$counts, Count_4$counts, Count_5$counts, Count_6$counts, Count_7$counts, Count_8$counts, Count_9$counts, Count_10$counts, Count_11$counts, Count_12$counts)
```
```{r}
coldata <- read.csv(file = "Data/coldata.csv", row.names=1, stringsAsFactors=FALSE)
```

Here deseq2 object is produced
```{r message = FALSE, warning = FALSE}
library ("dplyr")
library("ggplot2")
library("vsn")
library("DESeq2")
ddsMat <- DESeqDataSetFromMatrix(countData = countdata,
                                 colData = coldata,
                                 design = ~ Cell)
dds <- DESeq(ddsMat)

```


Compare different transformation methods for PCA
```{r message = FALSE, warning = FALSE}

vsd <- vst(ddsMat, blind = FALSE)
rld <- rlog(ddsMat, blind = FALSE)
ntd <- normTransform(ddsMat)

df <- bind_rows(
  as_data_frame(log2(counts(dds, normalized=TRUE)[, 1:2]+1)) %>%
         mutate(transformation = "log2(x + 1)"),
  as_data_frame(assay(vsd)[, 1:2]) %>% mutate(transformation = "vst"),
  as_data_frame(assay(rld)[, 1:2]) %>% mutate(transformation = "rlog"))
  
colnames(df)[1:2] <- c("x", "y")  

lvls <- c("log2(x + 1)", "vst", "rlog")
df$transformation <- factor(df$transformation, levels=lvls)

ggplot(df, aes(x = x, y = y)) + geom_hex(bins = 80) +
  coord_fixed() + facet_grid( . ~ transformation)  

```


Different PCA plots are produced

```{r message = FALSE, warning = FALSE,  fig.show="hold", out.width="50%"}
plotPCA(vsd, intgroup = "Cell", ntop = 500)
plotPCA(rld, intgroup = "Cell", ntop = 500)
plotPCA(ntd, intgroup = "Cell", ntop = 500)
```


 All of the PCA plots show a clear clustering of our samples. This confirms the samples have good quality


 SD against mean is calculated per type of transformation vsd results look better both with this plots and with PCA. vsd will be used to compare group distances in PCA environment
```{r message = FALSE, warning = FALSE,  fig.show="hold", out.width="50%"} 
meanSdPlot(assay(ntd))
meanSdPlot(assay(rld))
meanSdPlot(assay(vsd))

```


 Vsd appear to be the most accurate as we don't expect intra group variations

The following are heatmaps which give information ragarding the general distances between groups
```{r message = FALSE, warning = FALSE}
sampleDists <- dist(t(assay(vsd)))
library("pheatmap")
library("RColorBrewer")

sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <-  vsd@colData@listData[["Cell"]]
colnames(sampleDistMatrix) <- vsd@colData@listData[["Cell"]]
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)
```

Poisson distances
```{r message = FALSE, warning = FALSE}
library("PoiClaClu")
poisd <- PoissonDistance(t(counts(dds)))

samplePoisDistMatrix <- as.matrix( poisd$dd )
rownames(samplePoisDistMatrix) <- dds$Cell
colnames(samplePoisDistMatrix) <- dds$Cell
pheatmap(samplePoisDistMatrix,
         clustering_distance_rows = poisd$dd,
         clustering_distance_cols = poisd$dd,
         col = colors)
```



individual genes can be studied. Example provided 
```{r message = FALSE, warning = FALSE,  fig.show="hold", out.width="50%"}
#topGene <- rownames(res)[which.min(res$padj)]
#plotCounts(dds, gene = topGene, intgroup=c("Cell"))
plotCounts(dds, gene = "NFE2L2", intgroup=c("Cell"))
```

Heatmaps of Top Variable genes
```{r message = FALSE, warning = FALSE,  fig.show="hold", out.width="50%"}
library("genefilter")
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 20)


mat  <- assay(vsd)[ topVarGenes, ]
mat  <- mat - rowMeans(mat)
anno <- as.data.frame(colData(vsd)[, "Cell"])
pheatmap(mat)#, annotation_col = anno )
library( "gplots" )
library( "RColorBrewer" )
library( "genefilter" )
topVarGenes <- head( order( rowVars( assay(vsd) ), decreasing=TRUE ), 35 )

heatmap.2( assay(vsd)[ topVarGenes, ], scale="row", 
     trace="none", dendrogram="column", 
     col = colorRampPalette( rev(brewer.pal(9, "RdBu")) )(255))#,
     #ColSideColors = c( Control="gray", DPN="darkgreen", OHT="orange" )[
       # colData(vsd)$treatment ] )
```



Count box plot and dispersion plots. Box plot shows there is no outlier among our sampes. Dispersion plot is also typical. "the final estimates shrunk from the gene-wise estimates towards the fitted estimates. Some gene-wise estimates are flagged as outliers and not shrunk towards the fitted value, (this outlier detection is described in the manual page for estimateDispersionsMAP)"
```{r message = FALSE, warning = FALSE,  fig.show="hold", out.width="50%"}
par(mar=c(8,5,2,2))
boxplot(log10(assays(dds)[["cooks"]]), range=0, las=2)
plotDispEsts(dds)

```

(To be done on results
Rank of Cook wald statistic . We use Cook’s distance, which is a measure of how much the fitted coefficients would change if an individual sample were removed. This is carried out as automatic detection of count outliers


Calculation of DEG:plus log2fold change



```{r message = FALSE, warning = FALSE}
dds$Cell <- relevel(dds$Cell, ref = "A549")
dds <- DESeq(dds)
NCI.H460vsA549 <- results(dds, contrast = c("Cell", "NCI-H460", "A549"), alpha = 0.05)
NCI.H460vsA549shrunk <- lfcShrink(dds, coef =4, type = "apeglm")



Calu1vsA549 <-   results(dds, contrast = c("Cell", "Calu1", "A549"), alpha = 0.05)
Calu1vsA549shrunk <- lfcShrink(dds, coef =2, type = "apeglm")


NCIH358vsA549 <-   results(dds, contrast = c("Cell", "NCI-H358", "A549"), alpha = 0.05)
NCIH358vsA549shrunk <- lfcShrink(dds, coef =3, type = "apeglm")


dds$Cell <- relevel(dds$Cell, ref = "Calu1")
dds <- DESeq(dds)
NCIH460vsCalu1 <-   results(dds, contrast = c("Cell", "NCI-H460", "Calu1"), alpha = 0.05)
NCIH460vsCalu1shrunk <- lfcShrink(dds, coef =3, type = "apeglm")


NCIH358vsCalu1 <-   results(dds, contrast = c("Cell", "NCI-H358", "Calu1"), alpha = 0.05)
NCIH358vsCalu1shrunk <- lfcShrink(dds, coef =4, type = "apeglm")

dds$Cell <- relevel(dds$Cell, ref = "NCI-H358")
dds <- DESeq(dds)
NCIH358vsNCIH460 <-   results(dds, contrast = c("Cell", "NCI-H460", "NCI-H358"), alpha = 0.05)
NCIH358vsNCIH460shrunk <- lfcShrink(dds, coef =4, type = "apeglm")

```

```{r}
res <- NCI.H460vsA549

plot(metadata(res)$filterNumRej, 
     type="b", ylab="number of rejections",
     xlab="quantiles of filter")
lines(metadata(res)$lo.fit, col="red")
abline(v=metadata(res)$filterTheta)

```


will use shrinkage for now
Matrices
```{r message = FALSE, warning = FALSE}
DEG_NCI.H460vsA549shrunk <- as.data.frame(NCI.H460vsA549shrunk)  %>%  dplyr::filter(padj < 0.05 & log2FoldChange > 0.5)  %>%  dplyr::arrange(desc(log2FoldChange)) %>% write.csv(file = "DEG_NCI.H460vsA549shrunk.csv")

DEG_Calu1vsA549shrunk <- as.data.frame(Calu1vsA549shrunk)  %>%  dplyr::filter(padj < 0.05 & log2FoldChange > 0.5)  %>%  dplyr::arrange(desc(log2FoldChange)) %>% write.csv(file = "DEG_Calu1vsA549shrunk.csv")

DEG_NCIH358vsA549shrunk <- as.data.frame(NCIH358vsA549shrunk)  %>%  dplyr::filter(padj < 0.05 & log2FoldChange > 0.5)  %>%  dplyr::arrange(desc(log2FoldChange)) %>% write.csv(file = "DEG_NCIH358vsA549shrunk.csv")

DEG_NCIH460vsCalu1shrunk <- as.data.frame(NCIH460vsCalu1shrunk)  %>%  dplyr::filter(padj < 0.05 & log2FoldChange > 0.5)  %>%  dplyr::arrange(desc(log2FoldChange))%>% write.csv(file = "DEG_NCIH460vsCalu1shrunk.csv")

DEG_NCIH358vsCalu1shrunk <- as.data.frame(NCIH358vsCalu1shrunk)  %>%  dplyr::filter(padj < 0.05 & log2FoldChange > 0.5)  %>%  dplyr::arrange(desc(log2FoldChange)) %>% write.csv(file = "DEG_NCIH358vsCalu1shrunk.csv")

DEG_NCIH358vsNCIH460shrunk <- as.data.frame(NCIH358vsNCIH460shrunk)  %>%  dplyr::filter(padj < 0.05 & log2FoldChange > 0.5)  %>%  dplyr::arrange(desc(log2FoldChange)) %>% write.csv(file = "DEG_NCIH358vsNCIH460shrunk.csv")


```

```{r}
colSums(counts(dds))
```

```{r}
summary(NCI.H460vsA549)
```

```{r message = FALSE, warning = FALSE}
library(EnhancedVolcano)
vp1 <- EnhancedVolcano(NCI.H460vsA549,
lab = rownames(NCI.H460vsA549),
x = 'log2FoldChange',
y = 'padj', pCutoff = 0.05,
title = "NCI.H460vsA549")
print(vp1)
```

```{r message = FALSE, warning = FALSE}
library(EnhancedVolcano)
vp1 <- EnhancedVolcano(NCI.H460vsA549shrunk,
lab = rownames(NCI.H460vsA549shrunk),
x = 'log2FoldChange',
y = 'padj', pCutoff = 0.05,
title = "NCI.H460vsA549")
print(vp1)
```


Next will be Kegg 

```{r message = FALSE, warning = FALSE}
library(pathview)
library(gage)
library(gageData)
data(kegg.sets.hs)
data(sigmet.idx.hs)
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]
head(kegg.sets.hs, 3)
```
```{r  message = FALSE, warning = FALSE}
library(org.Hs.eg.db)
library(AnnotationDbi) 
NCI.H460vsA549shrunk$symbol = mapIds(org.Hs.eg.db,
                     keys=row.names(NCI.H460vsA549shrunk), 
                     column="SYMBOL",
                     keytype="SYMBOL",
                     multiVals="first")
NCI.H460vsA549shrunk$entrez = mapIds(org.Hs.eg.db,
                     keys=row.names(NCI.H460vsA549shrunk), 
                     column="ENTREZID",
                     keytype="SYMBOL",
                     multiVals="first")
NCI.H460vsA549shrunk$name =   mapIds(org.Hs.eg.db,
                     keys=row.names(NCI.H460vsA549shrunk), 
                     column="GENENAME",
                     keytype="SYMBOL",
                     multiVals="first")
NCI.H460vsA549shrunk$ENSEMBL =   mapIds(org.Hs.eg.db,
                     keys=row.names(NCI.H460vsA549shrunk), 
                     column="ENSEMBL",
                     keytype="SYMBOL",
                     multiVals="first")


head(NCI.H460vsA549shrunk, 10)
```

```{r message = FALSE, warning = FALSE}
foldchanges = NCI.H460vsA549shrunk$log2FoldChange
names(foldchanges) = NCI.H460vsA549shrunk$entrez
head(foldchanges)

```


```{r message = FALSE, warning = FALSE}
keggres = gage(foldchanges, gsets=kegg.sets.hs, same.dir=TRUE)
# Look at both up (greater), down (less), and statatistics.
lapply(keggres, head)

```



```{r message = FALSE, warning = FALSE}
keggrespathways = data.frame(id=rownames(keggres$greater), keggres$greater) %>% 
  tbl_df() %>% 
  filter(row_number()<=1000) %>% 
  .$id %>% 
  as.character()
#keggrespathways

keggresids = substr(keggrespathways, start=1, stop=8)
keggresids

```



```{r message = FALSE, warning = FALSE}
# Define plotting function for applying later
#plot_pathway = function(pid) pathview(gene.data=foldchanges, pathway.id=pid, species="hsa", new.signature=FALSE)
# plot multiple pathways (plots saved to disk and returns a throwaway list object)
#tmp = sapply(keggresids, function(pid) pathview(gene.data=foldchanges, pathway.id=pid, species="hsa"))
```




```{r}
#tmp
```

```{r message = FALSE, warning = FALSE}
library(clusterProfiler)
library(dplyr)
universe <-  NCI.H460vsA549shrunk$symbol
NCI.H460vsA549shrunk <- as.data.frame(NCI.H460vsA549shrunk)
sigGenes <- dplyr::filter(NCI.H460vsA549shrunk, padj < 0.05, log2FoldChange >1) #& !is.na(A549_NCIH460$symbol)

enrich_go <- enrichGO(
  gene= sigGenes$symbol,
  OrgDb = org.Hs.eg.db,
  keyType = "SYMBOL",
  ont = "ALL",
  universe = universe,
  qvalueCutoff = 0.05,
  pvalueCutoff = 0.05,
  readable=TRUE
)

enrich_go_tidy <- enrich_go %>% 
  slot("result") %>% 
  tibble::as_tibble() 
enrich_go_tidy
#pdf(file = "GOEgzample.pdf", width = 6, height = 8)
dotplot(enrich_go, showCategory=25)
#dev.off()
#emapplot(enrich_go)
x2 <- enrichplot::pairwise_termsim(enrich_go)
emapplot(x2,cex_label_category = 0.6, group_legend = TRUE)
```

```{r width = 6, height = 80}
enrich_go_tidy[,-3:-10]
```


Import 2 gene lists

plot heat map deg expression
or 
plot heat map of gene expression



use vsd because:
Why not raw counts? Two issues:

    Raw counts range is very large
    Variance increases with mean gene expression, this has impact on assessing the relationships.
```{r}
Gene_data_frame <- read.csv("Second_Alignment/Gene_list.csv")
```


```{r}
Gene_data_frame
```

```{r}
library("genefilter")


mat  <- assay(vsd)[ topVarGenes, ]
mat  <- mat - rowMeans(mat)
anno <- as.data.frame(colData(vsd)[, "Cell"])
pheatmap(mat)#, annotation_col = anno )
library( "gplots" )
library( "RColorBrewer" )
library( "genefilter" )
topVarGenes <- head( order( rowVars( assay(vsd) ), decreasing=TRUE ), 35 )

heatmap.2( assay(vsd)[ topVarGenes, ], scale="row", 
     trace="none", dendrogram="column", 
     col = colorRampPalette( rev(brewer.pal(9, "RdBu")) )(255))#,
```
annotation_col=
df <- as.data.frame(colData(dds)
```{r}
genes <- Gene_data_frame[,1]
genes <- as.vector.data.frame(genes)
pheatmap(assay(vsd)[genes,], cluster_rows=FALSE, show_rownames=FALSE,
     cluster_cols=FALSE )
```
isOR = grepl( "^OR",  xx)

where xx is, for instance, fData(cds)$SYMBOL, or whereever you have
stored the gene names. Then

  select = order( isOR, rowMeans(counts(cds)), decreasing=TRUE)[1:100]

will select you the top 100 most highly expressed genes among those
for which isOR is TRUE.
(More precisely, it will first take the most highly expressed genes
among those for which isOR is TRUE, until these are exhausted, then
continue with the ones for which it is FALSE.) Another option is thus

  select = order(rowMeans(counts(cds)), decreasing=TRUE)[1:100]
  select = intersect(select, which(isOR))





```{r}

isOR = grepl( "^OR",  genes)
select = order( isOR, rowVars( assay(vsd) ))#[1:100]
```

```{r}
library(viridis)
# Transform count data using the variance stablilizing transform
deseq2VST <- vsd

# Convert the DESeq transformed object to a data frame
deseq2VST <- assay(deseq2VST)
deseq2VST <- as.data.frame(deseq2VST)
deseq2VST$Gene <- rownames(deseq2VST)
head(deseq2VST)

# Keep only the significantly differentiated genes where the fold-change was at least 3
sigGenes <- genes
deseq2VST <- deseq2VST[deseq2VST$Gene %in% sigGenes,]

# Convert the VST counts to long format for ggplot2
library(reshape2)

# First compare wide vs long version
deseq2VST_wide <- deseq2VST
deseq2VST_long <- melt(deseq2VST, id.vars=c("Gene"))

 


head(deseq2VST_wide)
head(deseq2VST_long)

# Now overwrite our original data frame with the long format
deseq2VST <- melt(deseq2VST, id.vars=c("Gene"))

#for (x in deseq2VST$variable){
  
#if (deseq2VST$variable[x] == rownames(coldata[x,1])) deseq2VST$variable[x] <- coldata[x,1]
#}
deseq2VST$variable[deseq2VST$variable == rownames(coldata)] <- coldata[,1]
coldata$variable <- coldata$X
deseq2VST2 <- dplyr::left_join(deseq2VST, coldata, by = "variable")

# Make a heatmap
heatmap <- ggplot(deseq2VST2, aes(x=Cell, y=Gene, fill=value)) + geom_raster() + scale_fill_viridis(trans="sqrt") + theme(axis.text.x=element_text(angle=65, hjust=1), axis.text.y=element_text(angle=0, hjust=1), axis.ticks.y=element_blank())
heatmap
heatmap2 <- ggplot(deseq2VST2, aes(x=Gene, y=Cell, fill=value)) + geom_raster() + scale_fill_viridis(trans="sqrt") + theme(axis.text.x=element_text(angle=65, hjust=1), axis.text.y=element_text(angle=0, hjust=1), axis.ticks.y=element_blank()) + theme(panel.background = element_rect(fill = NA)) +labs(y = NULL)+labs(fill = "Expression")
heatmap2
heatmap3 <- ggplot(deseq2VST2, aes(x=Gene, y=X, fill=value)) + geom_raster() + scale_fill_viridis(trans="sqrt") + theme(axis.text.x=element_text(angle=65, hjust=1), axis.text.y=element_text(angle=0, hjust=1), axis.ticks.y=element_blank())
heatmap3
```
```{r}
library(viridis)
# Transform count data using the variance stablilizing transform
deseq2VST <- ntd

# Convert the DESeq transformed object to a data frame
deseq2VST <- assay(deseq2VST)
deseq2VST <- as.data.frame(deseq2VST)
deseq2VST$Gene <- rownames(deseq2VST)
head(deseq2VST)

# Keep only the significantly differentiated genes where the fold-change was at least 3
sigGenes <- genes
deseq2VST <- deseq2VST[deseq2VST$Gene %in% sigGenes,]

# Convert the VST counts to long format for ggplot2
library(reshape2)

# First compare wide vs long version
deseq2VST_wide <- deseq2VST
deseq2VST_long <- melt(deseq2VST, id.vars=c("Gene"))

 


head(deseq2VST_wide)
head(deseq2VST_long)

# Now overwrite our original data frame with the long format
deseq2VST <- melt(deseq2VST, id.vars=c("Gene"))

#for (x in deseq2VST$variable){
  
#if (deseq2VST$variable[x] == rownames(coldata[x,1])) deseq2VST$variable[x] <- coldata[x,1]
#}
deseq2VST$variable[deseq2VST$variable == rownames(coldata)] <- coldata[,1]
coldata$variable <- coldata$X
deseq2VST2 <- dplyr::left_join(deseq2VST, coldata, by = "variable")

# Make a heatmap
heatmap <- ggplot(deseq2VST2, aes(x=Cell, y=Gene, fill=value)) + geom_raster() + scale_fill_viridis(trans="sqrt") + theme(axis.text.x=element_text(angle=65, hjust=1), axis.text.y=element_text(angle=0, hjust=1), axis.ticks.y=element_blank())
heatmap
heatmap2 <- ggplot(deseq2VST2, aes(x=Gene, y=Cell, fill=value)) + geom_raster() +  scale_fill_viridis_c(option = "magma") + theme(axis.text.x=element_text(angle=65, hjust=1), axis.text.y=element_text(angle=0, hjust=1), axis.ticks.y=element_blank()) + theme(panel.background = element_rect(fill = NA)) +labs(y = NULL)+labs(fill = "Expression")
heatmap2
heatmap3 <- ggplot(deseq2VST2, aes(x=Legend, y=Gene, fill=value)) + geom_raster() + scale_fill_distiller(palette = "RdBu")+ theme(axis.text.x=element_text(angle=65, hjust=1), axis.text.y=element_text(angle=0, hjust=1), axis.ticks.y=element_blank())
heatmap3
```

change x1 x2 x3 bam with 1-1 1-2 1-3 and 2-1 2-2 2-3 etc with name of sample



```{r}
heatmap4 <- ggplot(deseq2VST2, aes(x=Legend, y=Gene, fill=value)) + geom_raster() + scale_fill_viridis_c(option = "magma") + theme(axis.text.x=element_text(angle=65, hjust=1), axis.text.y=element_text(angle=0, hjust=1), axis.ticks.y=element_blank())
heatmap4
```









```{r}
coldata$variable <- coldata$X
deseq2VST2 <- dplyr::left_join(deseq2VST, coldata, by = "variable")
```
df$conf_full[which(df$conference=='W')] <- 'West'
df$conf_full[which(df$conference=='E')] <- 'East'

```{r}
deseq2VST2$Legend <- NULL
deseq2VST2$Legend[which(deseq2VST2$X == "X1.bam") ] <- "A549"
deseq2VST2$Legend[which(deseq2VST2$X == "X2.bam") ] <- "A549 "
deseq2VST2$Legend[which(deseq2VST2$X == "X3.bam") ] <- "A549  "

deseq2VST2$Legend[which(deseq2VST2$X == "X4.bam") ] <- "NCI-H460"
deseq2VST2$Legend[which(deseq2VST2$X == "X5.bam") ] <- "NCI-H460 "
deseq2VST2$Legend[which(deseq2VST2$X == "X6.bam") ] <- "NCI-H460  "

deseq2VST2$Legend[which(deseq2VST2$X == "X7.bam") ] <- "Calu1"
deseq2VST2$Legend[which(deseq2VST2$X == "X8.bam") ] <- "Calu1 "
deseq2VST2$Legend[which(deseq2VST2$X == "X9.bam") ] <- "Calu1  "

deseq2VST2$Legend[which(deseq2VST2$X == "X10.bam") ] <- "NCI-H358"
deseq2VST2$Legend[which(deseq2VST2$X == "X11.bam") ] <- "NCI-H358 "
deseq2VST2$Legend[which(deseq2VST2$X == "X12.bam") ] <- "NCI-H358  "
```


```{r}
deseq2VST2$Legend <- NULL
deseq2VST2$Legend[which(deseq2VST2$X == "X1.bam") ] <- "1.1 A549"
deseq2VST2$Legend[which(deseq2VST2$X == "X2.bam") ] <- "1.2 A549"
deseq2VST2$Legend[which(deseq2VST2$X == "X3.bam") ] <- "1.3 A549"

deseq2VST2$Legend[which(deseq2VST2$X == "X4.bam") ] <- "2.1 NCI-H460"
deseq2VST2$Legend[which(deseq2VST2$X == "X5.bam") ] <- "2.2 NCI-H460"
deseq2VST2$Legend[which(deseq2VST2$X == "X6.bam") ] <- "2.3 NCI-H460"

deseq2VST2$Legend[which(deseq2VST2$X == "X7.bam") ] <- "3.1 Calu1"
deseq2VST2$Legend[which(deseq2VST2$X == "X8.bam") ] <- "3.2 Calu1"
deseq2VST2$Legend[which(deseq2VST2$X == "X9.bam") ] <- "3.3 Calu1"

deseq2VST2$Legend[which(deseq2VST2$X == "X10.bam") ] <- "4.1 NCI-H358"
deseq2VST2$Legend[which(deseq2VST2$X == "X11.bam") ] <- "4.2 NCI-H358"
deseq2VST2$Legend[which(deseq2VST2$X == "X12.bam") ] <- "4.3 NCI-H358"
```



```{r}
library("genefilter")
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 20)


mat  <- assay(vsd)[ topVarGenes, ]
mat  <- mat - rowMeans(mat)
anno <- as.data.frame(colData(vsd)[, "Cell"])
pheatmap(mat)#, annotation_col = anno )
library( "gplots" )
library( "RColorBrewer" )
library( "genefilter" )
topVarGenes <- head( order( rowVars( assay(vsd) ), decreasing=TRUE ), 35 )

heatmap.2( assay(vsd)[ topVarGenes, ], scale="row", 
     trace="none", dendrogram="column", 
     col = colorRampPalette( rev(brewer.pal(9, "RdBu")) )(255))#,
     #ColSideColors = c( Control="gray", DPN="darkgreen", OHT="orange" )[
       # colData(vsd)$treatment ] )
```
```{r}
library("pheatmap")
select <- order(rowMeans(counts(dds,normalized=TRUE)),
                decreasing=TRUE)[1:20]
df <- as.data.frame(colData(dds))


pheatmap(assay(rld)[select,], cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=FALSE, annotation_col=df)
```
sigGenes <- genes
deseq2VST <- deseq2VST[deseq2VST$Gene %in% sigGenes,]
```{r}
mat  <- counts(dds, normalized =T)
mat.z <- t(apply(mat,1,scale))
z <- mat.z[rownames(mat.z) %in% genes,]  
pheatmap(z)
```




Timp2, Timp3 and Timp4)

```{r}
DefaultAssay(LASTCombined) <- "RNA"
FeaturePlot(LASTCombined, "TIMP1")
FeaturePlot(LASTCombined, "TIMP2")
FeaturePlot(LASTCombined, "TIMP3")
FeaturePlot(LASTCombined, "TIMP4")
```



```{r}
pdf(file="TIMP.pdf", height = 10,width = 8)
DotPlot(LASTCombined, split.by = "Detail_Condition", features = c("TIMP1","TIMP2","TIMP3","TIMP4"), cols = "RdBu")
dev.off()
```



```{r}

```


```{r}

```



Heatmap generation:
variable_gene_heatmap <- function (vsd.obj, num_genes = 500, annotation, title = "") {
  brewer_palette <- "RdBu"
  # Ramp the color in order to get the scale.
  ramp <- colorRampPalette( RColorBrewer::brewer.pal(11, brewer_palette))
  mr <- ramp(256)[256:1]
  # get the stabilized counts from the vsd object
  stabilized_counts <- assay(vsd.obj)
  # calculate the variances by row(gene) to find out which genes are the most variable across the samples.
  row_variances <- rowVars(stabilized_counts)
  # get the top most variable genes
  top_variable_genes <- stabilized_counts[order(row_variances, decreasing=T)[1:num_genes],]
  # subtract out the means from each row, leaving the variances for each gene
  top_variable_genes <- top_variable_genes - rowMeans(top_variable_genes, na.rm=T)
  # replace the ensembl ids with the gene names
  gene_names <- annotation$Gene.name[match(rownames(top_variable_genes), annotation$Gene.stable.ID)]
  rownames(top_variable_genes) <- gene_names
  # reconstruct colData without sizeFactors for heatmap labeling
  coldata <- as.data.frame(vsd.obj@colData)
  coldata$sizeFactor <- NULL
  # draw heatmap using pheatmap
  pheatmap::pheatmap(top_variable_genes, color = mr, annotation_col = coldata, fontsize_col = 8, fontsize_row = 250/num_genes, border_color = NA, main = title)
}

variable_gene_heatmap(vsd, num_genes = 40, annotation = annotation)


```{r}
variable_gene_heatmap <- function (vsd.obj, num_genes = 500, annotation, title = "") {
  brewer_palette <- "RdBu"
  # Ramp the color in order to get the scale.
  ramp <- colorRampPalette( RColorBrewer::brewer.pal(11, brewer_palette))
  mr <- ramp(256)[256:1]
  # get the stabilized counts from the vsd object
  stabilized_counts <- assay(vsd.obj)
  # calculate the variances by row(gene) to find out which genes are the most variable across the samples.
  row_variances <- rowVars(stabilized_counts)
  # get the top most variable genes
  top_variable_genes <- stabilized_counts[order(row_variances, decreasing=T)[1:num_genes],]
  # subtract out the means from each row, leaving the variances for each gene
  top_variable_genes <- top_variable_genes - rowMeans(top_variable_genes, na.rm=T)
  # replace the ensembl ids with the gene names
  gene_names <- annotation$Gene.name[match(rownames(top_variable_genes), annotation$Gene.stable.ID)]
  rownames(top_variable_genes) <- gene_names
  # reconstruct colData without sizeFactors for heatmap labeling
  coldata <- as.data.frame(vsd.obj@colData)
  coldata$sizeFactor <- NULL
  # draw heatmap using pheatmap
  pheatmap::pheatmap(top_variable_genes, color = mr, annotation_col = coldata, fontsize_col = 8, fontsize_row = 250/num_genes, border_color = NA, main = title)
}

variable_gene_heatmap(vsd, num_genes = 40, annotation = annotation)
```

```{r}

```


do same with second list plus check for missing genes then obtain go terms and gene expression differences from the comparisons



```{r}

```





