---
title: "LungCancer"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




(base) lucamannino@d39150v1:~/R_Projects/BulkRNAseq/Data$ Exited GFM loop
Exited: command not found
(base) lucamannino@d39150v1:~/R_Projects/BulkRNAseq/Data$ for f in `ls -1 *_1.fastqc.gz | sed 's/_1.fastqc.gz//' `
> do
> hisat2 -x index_ReferenceGenome -1 ${f}_1.fastqc.gz -2 ${f}_2.fastqc.gz -S ${f}.bam
> done
Warning: Unsupported file format
24101345 reads; of these:
  24101345 (100.00%) were paired; of these:
    710681 (2.95%) aligned concordantly 0 times
    22601988 (93.78%) aligned concordantly exactly 1 time
    788676 (3.27%) aligned concordantly >1 times
    ----
    710681 pairs aligned concordantly 0 times; of these:
      155955 (21.94%) aligned discordantly 1 time
    ----
    554726 pairs aligned 0 times concordantly or discordantly; of these:
      1109452 mates make up the pairs; of these:
        511792 (46.13%) aligned 0 times
        542064 (48.86%) aligned exactly 1 time
        55596 (5.01%) aligned >1 times
98.94% overall alignment rate
Warning: Unsupported file format
24120258 reads; of these:
  24120258 (100.00%) were paired; of these:
    741223 (3.07%) aligned concordantly 0 times
    22700322 (94.11%) aligned concordantly exactly 1 time
    678713 (2.81%) aligned concordantly >1 times
    ----
    741223 pairs aligned concordantly 0 times; of these:
      165786 (22.37%) aligned discordantly 1 time
    ----
    575437 pairs aligned 0 times concordantly or discordantly; of these:
      1150874 mates make up the pairs; of these:
        528929 (45.96%) aligned 0 times
        567263 (49.29%) aligned exactly 1 time
        54682 (4.75%) aligned >1 times
98.90% overall alignment rate
Warning: Unsupported file format
24004924 reads; of these:
  24004924 (100.00%) were paired; of these:
    614287 (2.56%) aligned concordantly 0 times
    22682012 (94.49%) aligned concordantly exactly 1 time
    708625 (2.95%) aligned concordantly >1 times
    ----
    614287 pairs aligned concordantly 0 times; of these:
      123420 (20.09%) aligned discordantly 1 time
    ----
    490867 pairs aligned 0 times concordantly or discordantly; of these:
      981734 mates make up the pairs; of these:
        458343 (46.69%) aligned 0 times
        477778 (48.67%) aligned exactly 1 time
        45613 (4.65%) aligned >1 times
99.05% overall alignment rate
Warning: Unsupported file format
24079108 reads; of these:
  24079108 (100.00%) were paired; of these:
    667689 (2.77%) aligned concordantly 0 times
    22618360 (93.93%) aligned concordantly exactly 1 time
    793059 (3.29%) aligned concordantly >1 times
    ----
    667689 pairs aligned concordantly 0 times; of these:
      131072 (19.63%) aligned discordantly 1 time
    ----
    536617 pairs aligned 0 times concordantly or discordantly; of these:
      1073234 mates make up the pairs; of these:
        510010 (47.52%) aligned 0 times
        512114 (47.72%) aligned exactly 1 time
        51110 (4.76%) aligned >1 times
98.94% overall alignment rate
Warning: Unsupported file format
24008980 reads; of these:
  24008980 (100.00%) were paired; of these:
    618427 (2.58%) aligned concordantly 0 times
    22616309 (94.20%) aligned concordantly exactly 1 time
    774244 (3.22%) aligned concordantly >1 times
    ----
    618427 pairs aligned concordantly 0 times; of these:
      130613 (21.12%) aligned discordantly 1 time
    ----
    487814 pairs aligned 0 times concordantly or discordantly; of these:
      975628 mates make up the pairs; of these:
        442217 (45.33%) aligned 0 times
        485477 (49.76%) aligned exactly 1 time
        47934 (4.91%) aligned >1 times
99.08% overall alignment rate
Warning: Unsupported file format
24148662 reads; of these:
  24148662 (100.00%) were paired; of these:
    709429 (2.94%) aligned concordantly 0 times
    22685604 (93.94%) aligned concordantly exactly 1 time
    753629 (3.12%) aligned concordantly >1 times
    ----
    709429 pairs aligned concordantly 0 times; of these:
      155360 (21.90%) aligned discordantly 1 time
    ----
    554069 pairs aligned 0 times concordantly or discordantly; of these:
      1108138 mates make up the pairs; of these:
        507126 (45.76%) aligned 0 times
        549381 (49.58%) aligned exactly 1 time
        51631 (4.66%) aligned >1 times
98.95% overall alignment rate
Warning: Unsupported file format
24153576 reads; of these:
  24153576 (100.00%) were paired; of these:
    682813 (2.83%) aligned concordantly 0 times
    22602818 (93.58%) aligned concordantly exactly 1 time
    867945 (3.59%) aligned concordantly >1 times
    ----
    682813 pairs aligned concordantly 0 times; of these:
      123607 (18.10%) aligned discordantly 1 time
    ----
    559206 pairs aligned 0 times concordantly or discordantly; of these:
      1118412 mates make up the pairs; of these:
        526367 (47.06%) aligned 0 times
        539107 (48.20%) aligned exactly 1 time
        52938 (4.73%) aligned >1 times
98.91% overall alignment rate
Warning: Unsupported file format
24137400 reads; of these:
  24137400 (100.00%) were paired; of these:
    779951 (3.23%) aligned concordantly 0 times
    22541424 (93.39%) aligned concordantly exactly 1 time
    816025 (3.38%) aligned concordantly >1 times
    ----
    779951 pairs aligned concordantly 0 times; of these:
      152855 (19.60%) aligned discordantly 1 time
    ----
    627096 pairs aligned 0 times concordantly or discordantly; of these:
      1254192 mates make up the pairs; of these:
        598352 (47.71%) aligned 0 times
        596999 (47.60%) aligned exactly 1 time
        58841 (4.69%) aligned >1 times
98.76% overall alignment rate
Warning: Unsupported file format
24101551 reads; of these:
  24101551 (100.00%) were paired; of these:
    779707 (3.24%) aligned concordantly 0 times
    22497085 (93.34%) aligned concordantly exactly 1 time
    824759 (3.42%) aligned concordantly >1 times
    ----
    779707 pairs aligned concordantly 0 times; of these:
      164486 (21.10%) aligned discordantly 1 time
    ----
    615221 pairs aligned 0 times concordantly or discordantly; of these:
      1230442 mates make up the pairs; of these:
        557994 (45.35%) aligned 0 times
        612966 (49.82%) aligned exactly 1 time
        59482 (4.83%) aligned >1 times
98.84% overall alignment rate
Warning: Unsupported file format
24032052 reads; of these:
  24032052 (100.00%) were paired; of these:
    622781 (2.59%) aligned concordantly 0 times
    22694637 (94.43%) aligned concordantly exactly 1 time
    714634 (2.97%) aligned concordantly >1 times
    ----
    622781 pairs aligned concordantly 0 times; of these:
      119818 (19.24%) aligned discordantly 1 time
    ----
    502963 pairs aligned 0 times concordantly or discordantly; of these:
      1005926 mates make up the pairs; of these:
        476071 (47.33%) aligned 0 times
        487632 (48.48%) aligned exactly 1 time
        42223 (4.20%) aligned >1 times
99.01% overall alignment rate
Warning: Unsupported file format
24083727 reads; of these:
  24083727 (100.00%) were paired; of these:
    693046 (2.88%) aligned concordantly 0 times
    22700477 (94.26%) aligned concordantly exactly 1 time
    690204 (2.87%) aligned concordantly >1 times
    ----
    693046 pairs aligned concordantly 0 times; of these:
      149017 (21.50%) aligned discordantly 1 time
    ----
    544029 pairs aligned 0 times concordantly or discordantly; of these:
      1088058 mates make up the pairs; of these:
        500461 (46.00%) aligned 0 times
        541783 (49.79%) aligned exactly 1 time
        45814 (4.21%) aligned >1 times
98.96% overall alignment rate
Warning: Unsupported file format
24122351 reads; of these:
  24122351 (100.00%) were paired; of these:
    682711 (2.83%) aligned concordantly 0 times
    22726585 (94.21%) aligned concordantly exactly 1 time
    713055 (2.96%) aligned concordantly >1 times
    ----
    682711 pairs aligned concordantly 0 times; of these:
      133859 (19.61%) aligned discordantly 1 time
    ----
    548852 pairs aligned 0 times concordantly or discordantly; of these:
      1097704 mates make up the pairs; of these:
        521731 (47.53%) aligned 0 times
        532138 (48.48%) aligned exactly 1 time
        43835 (3.99%) aligned >1 times
98.92% overall alignment rate
(base) lucamannino@d39150v1:~/R_Projects/BulkRNAseq/Data$ 










```{r}
#define shared directory for RNAseq data
RNAseqDATADIR <- "QCResults/"
#list the fastq files in the raw data directory
dir(RNAseqDATADIR)


```
#QC

```{r}
fastqc(fq.dir = "Data", # FASTQ files directory
       qc.dir = "QCResults", # Results direcory
       threads = 40                    # Number of threads
       )
```


```{r}
library(fastqcr)

# Aggregating Multiple FastQC Reports into a Data Frame 
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

# Demo QC directory containing zipped FASTQC reports
qc.dir <- system.file("fastqc_results", package = "fastqcr")
qc <- qc_aggregate(RNAseqDATADIR)
qc

# Inspecting QC Problems
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

# See which modules failed in the most samples
qc_fails(qc, "module")
# Or, see which samples failed the most
qc_fails(qc, "sample")

# Building Multi QC Reports
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
qc_report(qc.dir, result.file = "multi-qc-report" )

# Building One-Sample QC Reports (+ Interpretation)
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
qc.file <- system.file("fastqc_results", "S1_fastqc.zip", package = "fastqcr")
qc_report(qc.file, result.file = "one-sample-report",
          interpret = TRUE)
```

```{r}
list.files(qc.dir)
```

```{r}
library(dplyr)
qc %>%
  select(sample, module, status) %>%    
  filter(status %in% c("WARN", "FAIL")) %>%
  arrange(sample)
```

```{r}
summary(qc)
```

```{r}
qc_stats(qc)
```

```{r}
qc_fails(qc, "module")
```


```{r}
# See which module warned in the most samples
qc_warns(qc, "module")

# See which modules failed or warned.
qc_problems(qc, "module")

qc_problems(qc, "module", compact = FALSE)

qc_problems(qc, "module",  name = "Per sequence GC content")

qc_problems(qc, "module",  name = "GC content")
```

```{r}
# See which samples had one or more failed modules
qc_fails(qc, "sample")

# See which samples had one or more module with failure or warning
qc_problems(qc, "sample", compact = FALSE)

#qc_problems(qc, "sample", name = "S1")
```

list.files(system.file("extdata/refs", package="Rhisat2"), pattern="\\.fa$")
#> [1] "chr1.fa" "chr2.fa" "chr3.fa"
refs <- list.files(system.file("extdata/refs", package="Rhisat2"), 
                   full.names=TRUE, pattern="\\.fa$")
td <- tempdir()
hisat2_build(references=refs, outdir=td, prefix="myindex", 
             force=TRUE, strict=TRUE, execute=TRUE)
```{r}

```

#After creating the index, reads can be aligned using the hisat2 wrapper function. Most commonly, the reads will be provided in fastq files (one file for single-end reads, two files for paired-end reads). The names of these files can be provided directly to the hisat2 function, either as a vector (for single-end reads) or as a list of length 2 (for paired-end reads, each list element corresponds to one mate). You also need to provide the path to the index generated by hisat2_build (the output directory combined with the prefix), and the output file name where the alignments should be written.


list.files(system.file("extdata/reads", package="Rhisat2"), 
           pattern="\\.fastq$")
#> [1] "reads1.fastq" "reads2.fastq"
reads <- list.files(system.file("extdata/reads", package="Rhisat2"),
                    pattern="\\.fastq$", full.names=TRUE)
hisat2(sequences=as.list(reads), index=file.path(td, "myindex"), 
       type="paired", outfile=file.path(td, "out.sam"), 
       force=TRUE, strict=TRUE, execute=TRUE)
```{r}

list.files(system.file("extdata/reads", package="Rhisat2"), 
           pattern="\\.fastq$")
```
```{r}
td <- "/home/lucamannino/ReferenceGenome"
```

```{r}
reads <- list.files(system.file("extdata/reads", package="Rhisat2"),
                    pattern="\\.fastq$", full.names=TRUE)
hisat2(sequences=as.list(reads), index=file.path(td, "myindex"), 
       type="paired", outfile=file.path(td, "out.sam"), 
       force=TRUE, strict=TRUE, execute=TRUE)
```


```{r}
library(Rsubread)

Count_1 <- featureCounts(files="Data/1.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_2 <- featureCounts(files="Data/2.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_3 <- featureCounts(files="Data/3.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_4 <- featureCounts(files="Data/4.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_5 <- featureCounts(files="Data/5.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_6 <- featureCounts(files="Data/6.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_7 <- featureCounts(files="Data/7.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_8 <- featureCounts(files="Data/8.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_9 <- featureCounts(files="Data/9.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_10 <- featureCounts(files="Data/10.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_11 <- featureCounts(files="Data/11.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")
Count_12 <- featureCounts(files="Data/12.bam",isPairedEnd=TRUE,annot.ext = "Data/data",isGTFAnnotationFile = TRUE,GTF.attrType = "gene_name")

```


                                 
                                 
```{r}
saveRDS(Count_1,file = "Count_1")
saveRDS(Count_2,file = "Count_2")
saveRDS(Count_3,file = "Count_3")
saveRDS(Count_4,file = "Count_4")
saveRDS(Count_5,file = "Count_5")
saveRDS(Count_6,file = "Count_6")
saveRDS(Count_7,file = "Count_7")
saveRDS(Count_8,file = "Count_8")
saveRDS(Count_9,file = "Count_9")
saveRDS(Count_10,file = "Count_10")
saveRDS(Count_11,file = "Count_11")
saveRDS(Count_12,file = "Count_12")

```                                 


```{r}
Count_1 <- readRDS(file = "Count_1")
Count_2 <- readRDS(file = "Count_2")
Count_3 <- readRDS(file = "Count_3")
Count_4 <- readRDS(file = "Count_4")
Count_5 <- readRDS(file = "Count_5")
Count_6 <- readRDS(file = "Count_6")
Count_7 <- readRDS(file = "Count_7")
Count_8 <- readRDS(file = "Count_8")
Count_9 <- readRDS(file = "Count_9")
Count_10 <- readRDS(file = "Count_10")
Count_11 <- readRDS(file = "Count_11")
Count_12 <- readRDS(file = "Count_12")



```
                                
                                 
```{r}
countdata <- data.frame(Count_1$counts, Count_2$counts, Count_3$counts, Count_4$counts, Count_5$counts, Count_6$counts, Count_7$counts, Count_8$counts, Count_9$counts, Count_10$counts, Count_11$counts, Count_12$counts)
```
```{r}
coldata <- read.csv(file = "Data/coldata.csv", row.names=1, stringsAsFactors=FALSE)
```



```{r}
library(DESeq2)
ddsMat <- DESeqDataSetFromMatrix(countData = countdata,
                                 colData = coldata,
                                 design = ~ Cell)
#dds <- DESeqDataSet(countdata, design = ~ Cell)
#dds <- DESeq(dds)
#vst <- assay(vst(dds))
```




```{r}
vsd <- vst(ddsMat, blind = FALSE)
head(assay(vsd), 3)
```
```{r}
rld <- rlog(ddsMat, blind = FALSE)
head(assay(rld), 3)
```

  
```{r}
library("dplyr")
library("ggplot2")

dds <- estimateSizeFactors(ddsMat)

df <- bind_rows(
  as_data_frame(log2(counts(dds, normalized=TRUE)[, 1:2]+1)) %>%
         mutate(transformation = "log2(x + 1)"),
  as_data_frame(assay(vsd)[, 1:2]) %>% mutate(transformation = "vst"),
  as_data_frame(assay(rld)[, 1:2]) %>% mutate(transformation = "rlog"))
  
colnames(df)[1:2] <- c("x", "y")  

lvls <- c("log2(x + 1)", "vst", "rlog")
df$transformation <- factor(df$transformation, levels=lvls)

ggplot(df, aes(x = x, y = y)) + geom_hex(bins = 80) +
  coord_fixed() + facet_grid( . ~ transformation)  
```


```{r}
ntd <- normTransform(dds)
library("vsn")
meanSdPlot(assay(ntd))
```

```{r}
meanSdPlot(assay(vsd))
```
```{r}
meanSdPlot(assay(rld))

```







Overall similarity between samples:
```{r}
sampleDists <- dist(t(assay(vsd)))
sampleDists
```



```{r}
library("pheatmap")
library("RColorBrewer")

sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <-  vsd@colData@listData[["Cell"]]
colnames(sampleDistMatrix) <- vsd@colData@listData[["Cell"]]
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)
```



```{r}
library("PoiClaClu")
poisd <- PoissonDistance(t(counts(dds)))

samplePoisDistMatrix <- as.matrix( poisd$dd )
rownames(samplePoisDistMatrix) <- dds$Cell
colnames(samplePoisDistMatrix) <- dds$Cell
pheatmap(samplePoisDistMatrix,
         clustering_distance_rows = poisd$dd,
         clustering_distance_cols = poisd$dd,
         col = colors)
```


```{r}
plotPCA(vsd, intgroup = "Cell")
```


Differential expression Analysis:

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
res
```
Pairing up each sample individually
```{r}
A549_NCIH460 <- results(dds, contrast = c("Cell", "A549", "NCI-H460"))

A549_Calu1 <-   results(dds, contrast = c("Cell", "A549", "Calu1"))

A549_NCIH358 <-   results(dds, contrast = c("Cell", "A549", "NCI-H358"))

Calu1_NCIH460 <-   results(dds, contrast = c("Cell", "Calu1", "NCI-H460"))

NCIH460_NCIH358 <-   results(dds, contrast = c("Cell", "NCI-H460", "NCI-H358"))

Calu1_NCIH358 <-   results(dds, contrast = c("Cell", "Calu1", "NCI-H358"))

```


```{r}
topGene <- rownames(res)[which.min(res$padj)]
plotCounts(dds, gene = topGene, intgroup=c("Cell"))
plotCounts(dds, gene = "PECAM1", intgroup=c("Cell"))
```

```{r}
library("ggbeeswarm")
geneCounts <- plotCounts(dds, gene = topGene, intgroup = "Cell",
                         returnData = TRUE)
ggplot(geneCounts, aes(x = Cell, y = count, color = Cell)) +
  scale_y_log10() +  geom_beeswarm(cex = 3)
```


```{r}
library("genefilter")
topVarGenes <- head(order(rowVars(assay(vsd)), decreasing = TRUE), 20)

mat  <- assay(vsd)[ topVarGenes, ]
mat  <- mat - rowMeans(mat)
anno <- as.data.frame(colData(vsd)[, "Cell"])
pheatmap(mat)#, annotation_col = anno )
```








similar as above
```{r}
library( "gplots" )
library( "RColorBrewer" )
library( "genefilter" )
topVarGenes <- head( order( rowVars( assay(vsd) ), decreasing=TRUE ), 35 )

heatmap.2( assay(vsd)[ topVarGenes, ], scale="row", 
     trace="none", dendrogram="column", 
     col = colorRampPalette( rev(brewer.pal(9, "RdBu")) )(255))#,
     #ColSideColors = c( Control="gray", DPN="darkgreen", OHT="orange" )[
       # colData(vsd)$treatment ] )
```




```{r}
DE_genes <- as.integer(dds$padj <= 0.05)
#names(DE_genes) <- rownames(dds)
```


Gene Ontology:

```{r}
A549_NCIH460_unshrunken <- results(dds, contrast = c("Cell", "A549", "NCI-H460"))


A549_NCIH460 <- lfcShrink(dds, contrast= c("Cell", "A549", "NCI-H460"), res=A549_NCIH460_unshrunken)

A549_Calu1 <-   results(dds, contrast = c("Cell", "A549", "Calu1"))

A549_NCIH358 <-   results(dds, contrast = c("Cell", "A549", "NCI-H358"))

Calu1_NCIH460 <-   results(dds, contrast = c("Cell", "Calu1", "NCI-H460"))

NCIH460_NCIH358 <-   results(dds, contrast = c("Cell", "NCI-H460", "NCI-H358"))

Calu1_NCIH358 <-   results(dds, contrast = c("Cell", "Calu1", "NCI-H358"))

```

```{r}
all_genes <- row.names(A549_NCIH460)
DE_genes <- all_genes[which(A549_NCIH460$padj<0.05)]
head(DE_genes)
```


```{r}
#BiocManager::install("goseq")
library(goseq)
```


Kegg Pathway

```{r}
library(pathview)
library(gage)
library(gageData)
data(kegg.sets.hs)
data(sigmet.idx.hs)
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]
head(kegg.sets.hs, 3)
```
```{r}
library(org.Hs.eg.db)
library(AnnotationDbi) 
A549_NCIH460$symbol = mapIds(org.Hs.eg.db,
                     keys=row.names(A549_NCIH460), 
                     column="SYMBOL",
                     keytype="SYMBOL",
                     multiVals="first")
A549_NCIH460$entrez = mapIds(org.Hs.eg.db,
                     keys=row.names(A549_NCIH460), 
                     column="ENTREZID",
                     keytype="SYMBOL",
                     multiVals="first")
A549_NCIH460$name =   mapIds(org.Hs.eg.db,
                     keys=row.names(A549_NCIH460), 
                     column="GENENAME",
                     keytype="SYMBOL",
                     multiVals="first")
A549_NCIH460$ENSEMBL =   mapIds(org.Hs.eg.db,
                     keys=row.names(A549_NCIH460), 
                     column="ENSEMBL",
                     keytype="SYMBOL",
                     multiVals="first")


head(A549_NCIH460, 10)
```


```{r}
foldchanges = A549_NCIH460$log2FoldChange
names(foldchanges) = A549_NCIH460$entrez
head(foldchanges)

```


```{r}
keggres = gage(foldchanges, gsets=kegg.sets.hs, same.dir=TRUE)
# Look at both up (greater), down (less), and statatistics.
lapply(keggres, head)

```



```{r}
keggrespathways = data.frame(id=rownames(keggres$greater), keggres$greater) %>% 
  tbl_df() %>% 
  filter(row_number()<=5) %>% 
  .$id %>% 
  as.character()
keggrespathways

keggresids = substr(keggrespathways, start=1, stop=8)
keggresids

```

```{r}
# Define plotting function for applying later
plot_pathway = function(pid) pathview(gene.data=foldchanges, pathway.id=pid, species="hsa", new.signature=FALSE)
# plot multiple pathways (plots saved to disk and returns a throwaway list object)
tmp = sapply(keggresids, function(pid) pathview(gene.data=foldchanges, pathway.id=pid, species="hsa"))
```




```{r}
tmp
```

Gene Ontology



```{r}
data(go.sets.hs)
data(go.subs.hs)
gobpsets = go.sets.hs[go.subs.hs$BP]
gobpres = gage(foldchanges, gsets=gobpsets, same.dir=TRUE)
lapply(gobpres, head)
```

```{r}
genesGO <- A549_NCIH460$padj < 0.05 & !is.na(A549_NCIH460$padj)
names(genesGO) <- A549_NCIH460$symbol
```


```{r}
#BiocManager::install("TxDb.Hsapiens.UCSC.hg38.knownGene")

library(goseq)
pwf <- nullp(genesGO, "hg38","geneSymbol")
```

```{r}
goseq_res <- goseq(pwf, "hg38","geneSymbol",test.cats="GO:BP")
head(goseq_res)
```
```{r}
library(clusterProfiler)
library(dplyr)
universe <-  !is.na(A549_NCIH460$symbol)
A549_NCIH460 <- as.data.frame(A549_NCIH460)
sigGenes <- dplyr::filter(A549_NCIH460, padj < 0.05, log2FoldChange > 1.5) #& !is.na(A549_NCIH460$symbol)

enrich_go <- enrichGO(
  gene= sigGenes$symbol,
  OrgDb = org.Hs.eg.db,
  keyType = "SYMBOL",
  ont = "ALL",
  universe = universe,
  qvalueCutoff = 0.05,
  pvalueCutoff = 0.05,
  readable=TRUE
)

enrich_go_tidy <- enrich_go %>% 
  slot("result") %>% 
  tibble::as_tibble() 
enrich_go_tidy

dotplot(enrich_go)
#emapplot(enrich_go)
x2 <- enrichplot::pairwise_termsim(enrich_go)
emapplot(x2,cex_label_category = 0.6, group_legend = TRUE)
```


